sqlmesh/utils/cache.py

```diff
@@ -119,7 +119,7 @@ def put(self, name: str, entry_id: str = "", *, value: T) -> None:
             raise SQLMeshError(f"Cache path '{self._path}' is not a directory.")
 
         with gzip.open(self._cache_entry_path(name, entry_id), "wb", compresslevel=1) as fd:
-            pickle.dump(value.dict(), fd)
+            pickle.dump(value.dict(exclude_none=False), fd)
```

Comment:
what is this

Reply:
Without this `OptimizedQueryCacheEntry(optimized_rendered_query=None)` will result in `{}` which somehow messes up deserialization later.